float32 fingerA
float32 fingerB
float32 fingerC
float32 scissor
